src/main/resources/swagger/ga4gh-tool-discovery.yaml

```diff
@@ -599,65 +616,41 @@ definitions:
     type: string
     description: >-
       The type of descriptor that represents this version of the tool (e.g. CWL,
-      WDL, or NFL).
+      WDL, or NFL). Note that these files can also include associated Docker/container files 
+      and test parameters that further describe a version of a tool
     enum:
       - CWL
       - WDL
       - NFL
-  ToolDescriptor:
+  FileWrapper:
     type: object
-    description: A tool descriptor is a metadata document that describes one or more tools.
+    description: A file provides content for one of; 
+      - A tool descriptor is a metadata document that describes one or more tools.
+      - A tool document that describes how to test with one or more sample test
+      JSON.
+      - A containerfile is a document that describes how to build a particular
+      container image. Examples include Dockerfiles for creating Docker images
+      and Singularity recipes for Singularity images
     required:
       - type
     properties:
       type:
         $ref: '#/definitions/DescriptorType'
```

Comment:
It's actually just the type of the descriptor as in the V1 ToolDescriptor
So its not necessarily applicable for containerfile for test files.

Reply:
Sorry, this somehow didn't get saved before submitting the last review:

Re: `FileWrapper` — the `type` property is required for *any* file, but is defined specifically by the `DescriptorType` schema. Because `DescriptorType` is an enum, then any file for which `type` is missing or has a value other than "CWL", "WDL", or "NFL" would be invalid.

In lieu of a better way to specify conditional enums for each file type, I'd vote for either leaving out `type` or keeping it as a generic string — e.g.,

```yaml
    required:
      - type
    properties:
      type:
        type: string
        description: The (sub)type/version of the file. For example, "CWL", "WDL", or "NFL" for descriptorfile; "JSON" or "YAML" for testfile; "Docker" or "Singularity" for containerfile.
      content:
        ...
      url:
        ...
```

Reply:
Sorry, this somehow didn't get saved before submitting the last review:

Re: `FileWrapper` — the `type` property is required for *any* file, but is defined specifically by the `DescriptorType` schema. Because `DescriptorType` is an enum, then any file for which `type` is missing or has a value other than "CWL", "WDL", or "NFL" would be invalid.

In lieu of a better way to specify conditional enums for each file type, I'd vote for either leaving out `type` or keeping it as a generic string — e.g.,

```yaml
    required:
      - type
    properties:
      type:
        type: string
        description: The (sub)type/version of the file. For example, "CWL", "WDL", or "NFL" for descriptorfile; "JSON" or "YAML" for testfile; "Docker" or "Singularity" for containerfile.
      content:
        ...
      url:
        ...
```

Reply:
Ah I see, I agree. 
I'll go with the simpler option, which is to remove it since I think the information is somewhat redundant actually.

Reply:
+1